### PR TITLE
Make fitDataset support multiple outputs

### PR DESCRIPTION
**Pull request by syt123450**
FEATURE
Make fitDataset support multiple outputs

This PR resolved TODO ([Handle case in which ys is a TensorMap](https://github.com/tensorflow/tfjs-layers/blob/master/src/engine/training_dataset.ts#L226)). Make .fitDataset() support multiple outputs.

PR includes three main parts:
* Check and Support `ys` as `TensorMap` in `standardizeDataIteratorOutput`.
* Make `dataset_fakes ` generate `ys` with multiple outputs.
* Add 8 test cases for this feature.

About new test cases added in this PR (including reference Python tf.keras code):

| Input | output | metric | validation | batchesPerEpoch |
| ----------- | ----------- | ----------- | ----------- | ----------- |
| 1 | 2 | 1 | no | with |
| 1 | 2 | 1 | no | without |
| 1 | 2 | 1 | tensor array | with |
| 1 | 2 | 1 | tensor array | without |
| 1 | 2 (missing 1) | 1 | no | with |
| 1 | 2 (missing 1) | 1 | no | without |
| 2 | 2 | 1 | no | with |
| 2 | 2 | 1 | no | without |

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/443)

### CODE REVIEW

**Comment by syt123450** [comment]
Thanks for the review, @caisq . Follow your suggestions, I did the modification. Let me know if PR needs further modifications, I will change them as soon as possible.

**Comment by caisq** [comment]
@syt123450 there are linter errors:  https://travis-ci.org/tensorflow/tfjs-layers/jobs/487788135

Can you fix them?

**Comment by syt123450** [comment]
@caisq Sorry for missing the lint check before. I have changed the code lint to ensure `yarn lint` pass.
